src/v/kafka/server/handlers/produce.cc

```diff
@@ -207,6 +209,22 @@ static partition_produce_stages partition_append(
     };
 }
 
+ss::future<produce_response::partition> finalize_request_with_error_code(
+  error_code ec,
+  std::unique_ptr<ss::promise<>>&& dispatch,
```

Comment:
prefer pass-by-value in most cases, like here. it leads to easier to understand behavior.